apps/mobile/src/features/TxHistory/components/TxHistoryList/TxHistoryList.tsx

```diff
@@ -1,62 +1,196 @@
-import React, { useMemo } from 'react'
+import React, { useMemo, useCallback } from 'react'
+import { useTheme, View, Text } from 'tamagui'
 import { Tabs } from 'react-native-collapsible-tab-view'
-import { useTheme, View } from 'tamagui'
-import { SafeListItem } from '@/src/components/SafeListItem'
-import { TransactionItem } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
-import { getTxHash, GroupedTxsWithTitle, groupTxsByDate } from '@/src/features/TxHistory/utils'
+import { getTxHash } from '@/src/features/TxHistory/utils'
 import { HistoryTransactionItems } from '@safe-global/store/gateway/types'
-import { renderItem } from '@/src/features/TxHistory/utils'
+import { TxGroupedCard } from '@/src/components/transactions-list/Card/TxGroupedCard'
+import { TxInfo } from '@/src/components/TxInfo'
 import { TransactionSkeleton, TransactionSkeletonItem } from '@/src/components/TransactionSkeleton'
 import { RefreshControl } from 'react-native'
 import { CircleSnail } from 'react-native-progress'
+import QuickCrypto from 'react-native-quick-crypto'
+import { formatWithSchema } from '@/src/utils/date'
+import { isDateLabel } from '@/src/utils/transaction-guards'
+import { TransactionItem } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { groupBulkTxs } from '@/src/utils/transactions'
 
 interface TxHistoryList {
   transactions?: HistoryTransactionItems[]
   onEndReached: (info: { distanceFromEnd: number }) => void
   isLoading?: boolean
+  isLoadingNext?: boolean
+  isLoadingPrevious?: boolean
   refreshing?: boolean
   onRefresh?: () => void
```

Comment:
According to how `TxHistoryList` is used these are all defined so the signature could be simplified and we wouldn't need a lot of the `!!` inside the component.